File: 926. Flip String to Monotone Increasing.py
# 926. Flip String to Monotone Increasing
class Solution:
    def minFlipsMonoIncr(self, s: str) -> int:
        ans = 0
        ones = 0
        for c in s:
            if c == "1":
                ones += 1
            else:
                ans = min(ones, ans + 1)
        return ans


# 
        # left_ones = 0
        # right_zeros = s.count("0")
        # ans = right_zeros
        # for c in s:
        #     if c == "0":
        #         right_zeros -= 1
        #     else:
        #         left_ones += 1
        #     ans = min(ans, left_ones + right_zeros)
        
        # return ans

File: 926. Flip String to Monotone Increasing.cpp
// 926. Flip String to Monotone Increasing


#include <string>
#include <algorithm>

class Solution {
    public:
        int minFlipsMonoIncr(std::string s) {
            int left_ones = 0;
            int right_zeros = count(s.begin(), s.end(), '0');
            int ans = right_zeros;
    
            for (char c : s) {
                if (c == '0') {
                    right_zeros--;
                } else {
                    left_ones += 1;
                }
                ans = std::min(ans, right_zeros + left_ones);
            }
    
            return ans;
        }
    };